website/blog/2019-01-23-jest-24-refreshing-polished-typescript-friendly.md

```diff
@@ -0,0 +1,104 @@
+
+![different types](/img/blog/24-different-types.png)
+
+Mock function not called
+
+![mock functions](/img/blog/24-mock-function.png)
+
+## Fixing old pain
+
+We've fixed a couple of really old issues in this release.
+
+The first one we'd like to highlight is `console.log` statements going missing. Jest intercepts and collects all logs in order to give you a stack trace to them, as well as provide them to reporters so you can handle them however you want. However, this has led to an issue where they have simply been missing in certain edge cases. Luckily for Jest 24, [@spion](https://twitter.com/spion) has [stepped up](https://github.com/facebook/jest/pull/6871) and fixed this issue. Thank you very much!
+
+The second one is an issue where Jest runs out of memory if the difference in serialization of expected and received value has a huge number of insertion changes (either unexpected because of mistake in test or defect in serializer or expected because of temporary failures during test-driven development). [@ittordepam](https://twitter.com/ittordepam) has [replaced](https://github.com/facebook/jest/pull/6961) the previous diffing algorithm with `diff-sequences` package, which should fix this issue because it uses the theoretical minimum amount of memory. It opens up possibility for word-diffs in the future, similar to what [git provides](https://git-scm.com/docs/git-diff#git-diff---word-diffltmodegt). Please see [this PR](https://github.com/facebook/jest/pull/4619) and don't hesitate to reach out if you want to help make that happen!
+
+## Other highlights
+
+- We have some improvements for `globalSetup` and `globalTeardown` as well - code transformation will be applied to them similar to `setupFiles` and they are now supported as part of `projects`.
+- You can [configure](https://github.com/facebook/jest/pull/6143) Jest's snapshot location, this is mainly useful if you are building tools which use Jest in a larger build process.
+- A quirk of Jest's CLI has been that while some flags and options have been camel cased (such as `runInBand`), others have not been (such as `no-cache`). In Jest 24, both are recognized, meaning you can write your CLI arguments however you want.
+- We've renamed `setupTestFrameworkScriptFile` to `setupFilesAfterEnv`, and made it into an array. We hope this will make it more obvious what the options is for. We have plans to further overhaul the configuration in the next major, see the paragraph in the section below.
+- To reduce the amount of magic Jest performs to “just work™”, in this release we decided to drop automatic injection of `regenerator-runtime` for transpiled async code to work. It's not always necessary (e.g. in Node 6+) and we believe it's the user's responsibility to handle it, which is already done if you use `@babel/preset-env` and `@babel/runtime` with correctly set up `targets` configuration. Please see our [Using Babel docs](/docs/en/getting-started#using-babel) for more information.
```

Comment:
```suggestion
- To reduce the amount of magic Jest performs to “just work™”, in this release we decided to drop automatic injection of `regenerator-runtime`, which is sometimes used in compiled async code. Including `regenerator-runtime` is not always necessary and we believe it's the user's responsibility to include it if it's needed. If you use `@babel/preset-env` with `targets` set to a modern Node version (Node 7.10.1+) you will not need to include it. Please see our [Using Babel docs](/docs/en/getting-started#using-babel) for more information.
```

Reply:
Nice, I like it! Only nit is that it's node Node 7 (that's for `async`) - you only need `regenerator-runtime` if you're on an older version than 4 (which itself has been EOL for almost a year). https://node.green/#ES2015-functions-generators. Node v0.12, which was when it was needed in Node, has been EOL since 2016

Reply:
You're right that 7 isn't the right version. I just tested it out, Babel uses `regeneratorRuntime` in Node 5.12.0 and lower.

Reply:
Changed back to "e.g. Node 6+".

Reply:
Thank you!